src/Domain/Wallet.cs

```diff
@@ -0,0 +1,256 @@
+﻿using System;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+using dotnetstandard_bip39;
+using Erdcsharp.Cryptography;
+using Erdcsharp.Domain.Helper;
+using Org.BouncyCastle.Crypto.Generators;
+using Org.BouncyCastle.Crypto.Parameters;
+using Org.BouncyCastle.Crypto.Signers;
+
+namespace Erdcsharp.Domain
+{
+    public class Wallet
+    {
+        private readonly        byte[]                   _privateKey;
+        private readonly        byte[]                   _publicKey;
+        private const           string                   HdPrefix = "m/44'/508'/0'/0'";
+        private static readonly RNGCryptoServiceProvider RngCsp   = new RNGCryptoServiceProvider();
+
+        public Wallet(string privateKeyHex)
+            : this(Converter.FromHexString(privateKeyHex))
+        {
+        }
+
+        /// <summary>
+        /// Get the account wallet
+        /// </summary>
+        /// <returns>Account</returns>
+        public Account GetAccount()
+        {
+            return new Account(Address.FromBytes(_publicKey));
+        }
+
+        /// <summary>
+        /// Build a wallet
+        /// </summary>
+        /// <param name="privateKey">The private key</param>
+        public Wallet(byte[] privateKey)
+        {
+            var privateKeyParameters = new Ed25519PrivateKeyParameters(privateKey, 0);
+            var publicKeyParameters  = privateKeyParameters.GeneratePublicKey();
+            _publicKey  = publicKeyParameters.GetEncoded();
+            _privateKey = privateKey;
+        }
+
+        /// <summary>
+        /// Derive a wallet from Mnemonic phrase
+        /// </summary>
+        /// <param name="mnemonic">The mnemonic phrase</param>
+        /// <param name="accountIndex">The account index, default 0</param>
+        /// <returns>Wallet</returns>
+        public static Wallet DeriveFromMnemonic(string mnemonic, int accountIndex = 0)
+        {
+            try
+            {
+                var bip39   = new BIP39();
+                var seedHex = bip39.MnemonicToSeedHex(mnemonic, "");
+
+                var hdPath     = $"{HdPrefix}/{accountIndex}'";
+                var kv         = DerivePath(hdPath, seedHex);
+                var privateKey = kv.Key;
+                return new Wallet(privateKey);
+            }
+            catch (Exception ex)
+            {
+                throw new Exception("CannotDeriveKeysException", ex);
+            }
+        }
+
+        /// <summary>
+        /// Derive a wallet from a KeyFile and a password
+        /// </summary>
+        /// <param name="keyFile">The KeyFile</param>
+        /// <param name="secretPassword">The password</param>
+        /// <returns>Wallet</returns>
+        public static Wallet DeriveFromKeyFile(KeyFile keyFile, string secretPassword)
```

Comment:
`DecryptFromKeyFile`? `Derive` isn't the correct verb here. CC: @ccorcoveanu.

Reply:
Recently, @ccorcoveanu  did a nice refactor in erdjs and created this package:

https://github.com/ElrondNetwork/elrond-sdk-erdjs/tree/main/src/crypto.

Perhaps, in a future PR, we can adjust `erdcsharp` to better match `erdjs`'s design?